### ISSUE TITLE
Tagged data seems not to work as expected

### ISSUE THREAD

**Issue by hansl** [anweiss/cddl#173]
The following CDDL:

```cddl
start = [ * help ]

help = #6.123(bstr)
```

With the following CBOR (diag notation, hex is `81D87B4100`):

```cbor
[ 123(h'00') ]
```

Does not validate using this library. The Ruby `cddl` gem validates it correctly. The errors are as follow:

```
$ cddl validate --cddl 2.cddl --cbor 2.cbor                                                                                                                                                                                                          
[ERROR] Validation of "5.cbor" failed: error validating group entry associated with rule "help" at cbor location : expected tagged data #6.123("hello"), got Array([Tag(123, Text("hello"))])
```

Edit: I just tested with the main branch and I get the same result.
Edit2: further simplified the CDDL and CBOR.

**Comment by hansl** [comment]
Note that the CDDL validates properly, the CBOR does not.

**Comment by hansl** [comment]
Creating some tests for the CBOR validation, it seems like maps and arrays don't properly visit the CBOR itself, instead staying at the root of the array/map. I'm still investigating and may come up with a PR to fix, as this is a very important tool for us.

**Comment by anweiss** [comment]
Fixed in #203. Re-open if needed.
